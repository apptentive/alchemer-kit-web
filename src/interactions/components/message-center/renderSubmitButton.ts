/**
 * Renders the submit button.
 *
 * @param {string} buttonText - The text to use for the button content
 * @param {Function} onSubmit - The handler for the click event on the button
 * @returns {HTMLDivElement} - The button container DOM Node.
 */
const renderSubmitButton = (buttonText: string, onSubmit: VoidFunction) => {
  const submitButton = document.createElement('button');
  submitButton.type = 'button';
  submitButton.classList.add('apptentive-button--primary');
  submitButton.textContent = buttonText;
  submitButton.addEventListener('click', onSubmit);

  const errors = document.createElement('apptentive-errors');
  errors.className = 'apptentive-errors';
  errors.textContent = '';
  errors.style.display = 'none';

  const buttonWrapper = document.createElement('div');
  buttonWrapper.classList.add('apptentive-message-center__actions');
  buttonWrapper.append(submitButton);
  buttonWrapper.append(errors);

  return buttonWrapper;
};

export default renderSubmitButton;
