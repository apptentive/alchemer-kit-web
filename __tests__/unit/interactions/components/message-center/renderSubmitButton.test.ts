import renderSubmitButton from '../../../../../src/interactions/components/message-center/renderSubmitButton';

const _buttonText = 'Click Me!';
const _onSubmit = jest.fn();

describe('renderSubmitButton', () => {
  test('properly renders submit button', () => {
    const output = renderSubmitButton(_buttonText, _onSubmit);

    expect(output).toMatchSnapshot();
  });

  test('properly attaches click event', () => {
    const output = renderSubmitButton(_buttonText, _onSubmit);
    const button = output.querySelector<HTMLButtonElement>('.apptentive-button--primary')!;
    button.click();

    expect(_onSubmit).toHaveBeenCalledTimes(1);
    _onSubmit.mockClear();
  });
});
