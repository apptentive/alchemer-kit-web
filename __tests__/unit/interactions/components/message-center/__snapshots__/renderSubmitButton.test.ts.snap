// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderSubmitButton properly renders submit button 1`] = `
<div
  class="apptentive-message-center__actions"
>
  <button
    class="apptentive-button--primary"
    type="button"
  >
    Click Me!
  </button>
  <apptentive-errors
    class="apptentive-errors"
    style="display: none;"
  />
</div>
`;
